Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.cpp

```diff
@@ -21,17 +21,29 @@ namespace ScriptCanvas
     static constexpr int MaxMessageLength = 4096;
     static constexpr const char ScriptCanvasAutoGenRegistrationWarningMessage[] = "[Warning] Registry name %s is occupied already, ignore AutoGen registry registration.\n";
 
+    static AZ::EnvironmentVariable<AutoGenRegistryManager> g_autogenRegistry;
+
     AutoGenRegistryManager::~AutoGenRegistryManager()
     {
         m_registries.clear();
     }
 
     AutoGenRegistryManager* AutoGenRegistryManager::GetInstance()
     {
-        // Use static object so each module will keep its own registry collection
-        static AutoGenRegistryManager s_autogenRegistry;
+        // Look up variable in AZ::Environment first
+        // This is need if the Environment variable was already created
+        if (!g_autogenRegistry)
+        {
+            g_autogenRegistry = AZ::Environment::FindVariable<AutoGenRegistryManager>(ScriptCanvasAutoGenRegistryName);
+        }
+
+        // Create the environment variable in this memory space if it has not been found
```

Comment:
"this memory space" isn't really correct anymore. The per-static-library copies of the `g_autogenRegistry` `EnvironmentVariable` merely hold a pointer to memory that is allocated by the `Environment` instance, which is in the memory space of the `O3DEKernel` library.

Reply:
thanks for correcting, will update